rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+is secure and authenticated. Before deployment, the contract code can be audited and verified to ensure it functions correctly. Once on the blockchain, 
+the immutable nature of the code prevents unauthorized modifications, further ensuring the integrity of the system.
+
+Running smart contracts requires computational resources, which are compensated through gas fees. These fees vary depending on the complexity of the operations within the contract. 
+More resource-intensive contracts incur higher fees, which helps manage the computational load on the blockchain network.
+
+The workflow of a smart contract begins with its development, where developers code the contract using languages like Rust. Once the code is compiled and deployed to the blockchain, 
+it becomes a permanent part of the network. Users can then interact with the contract by sending transactions that invoke specific functions. The contract checks whether the 
+required conditions are met, and if so, it automatically executes the specified actions, such as transferring tokens or updating data on the blockchain.
+
+The benefits of smart contracts are numerous. They eliminate the need for intermediaries by providing a system where trust is built into the code itself. 
+This not only reduces costs but also increases efficiency by automating repetitive processes. The inherent security of smart contracts, combined with their 
+transparency—where every action is recorded and visible on the blockchain—makes them a powerful tool for ensuring accountability and trust in decentralized systems. 
+They can be ideal for managing decentralized autonomous organizations (DAOs), where governance decisions are automated through coded rules.
+
+By integrating smart contracts, the Nescience architecture offers a highly versatile, secure, and transparent framework that can support a wide range of applications 
+across various industries, from finance to governance, supply chains, and more.
+
+## 2. General Execution Overview
+---
+This section explains the execution process within the Nescience architecture, providing an overview of how it works from start to finish. 
+It outlines the steps involved in each execution type, guiding the reader through the entire process from user interaction to completion.
+
+The process begins when a user initiates a transaction by invoking a smart contract (SC). This invocation involves selecting at least one of 
+the four execution types: public, private, shielded, or deshielded. The choice of execution type determines how data will be read from and written to the blockchain, 
+affecting the transaction's privacy and security levels. Each execution type caters to different privacy needs, allowing the user to tailor the transaction according 
+to their specific requirements, whether it be full transparency or complete confidentiality.
+
+![general](/img/general.png)
+
+
+#### User Actions:
+
+**Step 1**: **Smart Contract Selection and Input Creation**
```

Comment:
Replace:
"Smart Contract Selection and Input Creation"

With:
"Smart contract selection and input creation"